### PR TITLE
test=release/1.4 cherry-pick (#16845)

### PR DESCRIPTION
**Pull request by liupluswei**
* optimize lstmp and sample_logits op, test=develop

* update op_use_default_grad_op_maker.spec, test=develop

* delete useless file,test=develop

* append 0 to dim variable to avoid memory reusage, test=develop

### CODE REVIEW

**Review by xuezhong — APPROVED** [review]
LGTM
